stackslib/src/chainstate/stacks/db/blocks.rs

```diff
@@ -6330,23 +6330,32 @@ impl StacksChainState {
         StacksChainState::can_admit_mempool_semantic(tx, is_mainnet)?;
 
         let conf = self.config();
-        let staging_height =
-            match self.get_stacks_block_height(current_consensus_hash, current_block) {
-                Ok(Some(height)) => height,
-                Ok(None) => {
-                    if *current_consensus_hash == FIRST_BURNCHAIN_CONSENSUS_HASH {
-                        0
-                    } else {
-                        return Err(MemPoolRejection::NoSuchChainTip(
-                            current_consensus_hash.clone(),
-                            current_block.clone(),
-                        ));
-                    }
-                }
-                Err(_e) => {
-                    panic!("DB CORRUPTION: failed to query block height");
+        // First check if this transaction is valid against the Nakamoto chain tip
+        let mut staging_height = NakamotoChainState::get_nakamoto_block_height(
+            self.db(),
+            current_consensus_hash,
+            current_block,
+        )
+        .expect("DB CORRUPTION: failed to query block height");
```

Comment:
I think it's better to not panic here:

```suggestion
        .map_err(|e| {
            warn!("DB Error during mempool admission: {e:?}");
            MemPoolRejection::NoSuchChainTip(
                        current_consensus_hash.clone(),
                        current_block.clone(),
             )
        })?; 
```

Reply:
Can do :)